wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -822,6 +847,27 @@ instance ToSchema TransactionDirection where
             & type_ .~ SwaggerString
             & enum_ ?~ ["outgoing", "incoming"]
 
+data TransactionStatus
+    = TxPending
+    | TxFailed
+    | TxOk
```

Comment:
I think we _might_ need to be a bit more exhaustive here, as historically the transaction status "finite state machine" has been subject of debate and gripe. The crux of the problem is that a `CPtxCondition` is reified into an internal `PtxCondition`, so removing states (like we did here for `TransactionStatus`) is something we might not afford at this stage, I think:

``` haskell
data PtxCondition
    = PtxApplying PtxPoolInfo         -- ^ Is waiting to be applyed
    | PtxInNewestBlocks PtxBlockInfo  -- ^ Recently appeared in block.
    | PtxPersisted                    -- ^ Transaction is ~guaranteed to remain
                                      --   in blockchain
                                      --   (with up to *high* assurance level)
    | PtxWontApply Text PtxPoolInfo   -- ^ Can't be applyed and requires user's
                                      --   input to reform tx
    | PtxCreating PtxPoolInfo         -- ^ Transaction is at its creation cycle
                                      --   (this constructor is at bottom for
                                      --   the sake of SafeCopy instance's
                                      --   backward-compatibility)
    deriving (Eq, Show)
```

I suspect we might need to spend some time thinking how to reify this into V1. Definitely we want things to look and feel cleaner for V1 without giving up useful information V0-users might expect to find or might be relying upon.

Reply:
We could just expose `type TransactionStatus = V1 PtxCondition` ? Then when we've migrated, we can replace the underlying type and keep the representation the same.

Reply:
Alternatively, I can remove the transaction status from this PR and we can create a new story for it, which provides much more time for detailed discussion and design :smile: 

At the least, I can replace the enum-sum type with a more rich tag-based encoding, like:

```
{
  tag: "transaction status constructor tag",
  obj: {
    // the object's fields according to tag
  }
}
```

For now, `tag` would have the string of the sum type, and `obj` would be empty (or null?) to indicate that we don't have extra fields. When we add new fields, we can do so in a non-breaking manner.

Reply:
@parsonsmatt Uhm, that's a thorny question 🤔 I like the first proposal slightly a bit more, just because tag-based encoding is usually less pleasant for humans, and we are trying to hit that sweet spot between humans & machines 😁 

What about exposing `V1 PtxCondition` as you propose but also analyse the transaction status story as part of next sprint? (I think this might even deserve a dedicated user story in YouTrack).

Reply:
Sounds good! I'll do `V1 PtxCondition` and delegate to it's Aeson instance for now.